### ISSUE TITLE
Http client keeps logging at ERROR level upon retries

### ISSUE THREAD

**Issue by alexmnyc** [akka/akka-http#3067]
Default http client pool retries with backoff. However, there should not be an error level logging for retriable exceptions which is completely misleading

```
ERROR akka.actor.ActorSystemImpl akka.actor.ActorSystemImpl((my-service) - Outgoing request stream error
[service] akka.stream.StreamTcpException: Tcp command [Connect(domain.test.com:443,None,List(),Some(10 seconds),true)] failed because of java.net.UnknownHostException: 
```

at the very least it should be at the WARN level but not ERROR. Better, yet, there should be a setting to log or not retry errors and at which level. The final error obviously should be handled by the callee

**Comment by alexmnyc** [comment]
Looks like it was related to default http client pool retries and backoff, i wasn't patient enough to see the end of it. However, there should not be an error level logging for retriable exception which is completely misleading!!!!

```
ERROR akka.actor.ActorSystemImpl akka.actor.ActorSystemImpl((my-service) - Outgoing request stream error
[service] akka.stream.StreamTcpException: Tcp command [Connect(unity.test.awstrp.net:443,None,List(),Some(10 seconds),true)] failed because of java.net.UnknownHostException: 
```

at the very least it should be at the WARN level but not ERROR! Better, yet, there should be a setting to log or not retry errors and at which level. The final error obviously should be handled by the end user

**Comment by jrudolph** [comment]
Yes, that's annoying indeed. It has already been fixed in master (#2905) and release-10.1 branch and is waiting for releases. Can you try if it is silent with `10.2.0-M1`?
